### PR TITLE
Dichotomy keymap by beat

### PR DESCRIPTION
**Pull request by dr-BEat**
Adding a keymap for the Dichotemy keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dichotomy/keymaps/beat/keymap.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#include QMK_KEYBOARD_H
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _MOUSE 8
+#define _ADJUST 16
```
Comment (fauxpark) [inline keyboards/dichotomy/keymaps/beat/keymap.c]: Non-consecutive layer numbers will generate extra empty layers in between, unnecessarily increasing flash usage. The better way to define layers is with an enum, eliminating the need to number them manually:

```suggestion
enum layer_names {
    _QWERTY,
    _LOWER,
    _RAISE,
    _MOUSE,
    _ADJUST
};
```

File: keyboards/dichotomy/keymaps/beat/keymap.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  [_MOUSE] = LAYOUT( /* Mouse layer, including buttons for clicking. */
+  _______, _______, _______, _______, _______, _______,        KC_VOLU, KC_HOME, KC_WH_U, KC_PGUP, _______, _______,
+  _______, _______, MS_BTN2, MS_BTN3, MS_BTN1, _______,        KC_VOLD, MS_BTN1, MS_BTN3, MS_BTN2, _______, _______,
+  _______, _______, _______, _______, _______, _______,        KC_HOME, KC_END,  KC_WH_D, KC_PGDN, _______, _______,
+                             _______, KC_UP,   _______,        _______, KC_UP,   _______,
+                    _______, KC_LEFT, KC_DOWN, KC_RGHT,        KC_LEFT, KC_DOWN, KC_RGHT, _______
+  )
+};
+
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (fauxpark) [inline keyboards/dichotomy/keymaps/beat/keymap.c]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: keyboards/dichotomy/config.h [hunk 3]
```diff
@@ -41,6 +41,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #define ONESHOT_TIMEOUT 500
 
+
+#define TAPPING_TERM 200
+#define TAPPING_TOGGLE 2
```
Comment (fauxpark) [inline keyboards/dichotomy/config.h]: This should really go in your keymap level config.h.

Comment (dr-BEat) [reply keyboards/dichotomy/config.h]: Thanks for the tip! I did not know you could have a config on keymaps itself.
I updated the PR

File: keyboards/dichotomy/keymaps/beat/keymap.c (hunk 2 above)
Comment (dr-BEat) [reply keyboards/dichotomy/keymaps/beat/keymap.c]: I updated the PR

File: keyboards/dichotomy/keymaps/beat/keymap.c (hunk 1 above)
Comment (dr-BEat) [reply keyboards/dichotomy/keymaps/beat/keymap.c]: Good point. I updated the PR

File: keyboards/dichotomy/keymaps/beat/keymap.c [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+enum layer_names {
+    _QWERTY,
+    _LOWER,
+    _RAISE,
+    _MOUSE,
+    _ADJUST
+};
+
+enum dichotomy_keycodes
+{
```
Comment (drashna) [inline keyboards/dichotomy/keymaps/beat/keymap.c]: Just a nit pick on formatting: 
```suggestion
enum dichotomy_keycodes {
```

Comment (dr-BEat) [reply keyboards/dichotomy/keymaps/beat/keymap.c]: I missed that one, thanks!

File: keyboards/dichotomy/dichotomy.c [hunk 5]
```diff
@@ -63,10 +63,11 @@ void pointing_device_task(void){
 }
 
 void led_init(void) {
-	DDRD  |= (1<<1);
+    DDRD |= (1<<1);
+    DDRF |= (1<<5) | (1<<6);
```
Comment (noroadsleft) [inline keyboards/dichotomy/dichotomy.c]:
```suggestion
    setPinOutput(D1);
    setPinOutput(F5);
    setPinOutput(F6);
```

File: keyboards/dichotomy/dichotomy.c [hunk 6]
```diff
@@ -63,10 +63,11 @@ void pointing_device_task(void){
...
+
 	PORTD |= (1<<1);
```
Comment (noroadsleft) [inline keyboards/dichotomy/dichotomy.c]:
```suggestion
```

File: keyboards/dichotomy/dichotomy.c [hunk 7]
```diff
@@ -63,10 +63,11 @@ void pointing_device_task(void){
...
-	DDRF  |= (1<<4) | (1<<5);
-	PORTF |= (1<<4) | (1<<5);
+	PORTF |= (1<<5) | (1<<6);
```
Comment (noroadsleft) [inline keyboards/dichotomy/dichotomy.c]:
```suggestion
    writePinHigh(D1);
    writePinHigh(F5);
    writePinHigh(F6);
```

File: keyboards/dichotomy/dichotomy.c (hunk 5 above)
Comment (dr-BEat) [reply keyboards/dichotomy/dichotomy.c]: Another thing learned. 👍

File: keyboards/dichotomy/dichotomy.c (hunk 7 above)
Comment (dr-BEat) [reply keyboards/dichotomy/dichotomy.c]: Done

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**